preview-env/create/action.yml

```diff
@@ -69,9 +69,40 @@ runs:
   - name: Deploy
     shell: bash
     run: >-
-      argocd app create ${{ inputs.argocd_arguments }}
+      if argocd app create ${{ inputs.argocd_arguments }}; then
+        echo "Creating argo cd app with name ${{ inputs.app_name }}"
 
-      argocd app sync ${{ inputs.app_name }} --async --force || true
+        # Record the start time
+        start_time=$(date +%s)
+
+        # Wait for the app to be registered
+        while true; do
+          echo "Printing app info: "
+          app_info=$(argocd app get ${{ inputs.app_name }} --output json 2>/dev/null)
+          echo "$app_info"
+
+          if [ $? -eq 0 ]; then
```

Comment:
$? always looks at the last executed command, in that case `echo "$app_info"`.
echo is typically always executing successfully, doesn't matter whether `app_info=$(...)` will fail or not, it won't influence the echo command.

At least I tried it out and app_info was having a return code 20 but the following echo would just print nothing and be a 0

Reply:
true echo is also a command 🤦‍♀️

Reply:
I think it makes sense to check that app_info is not empty and that the exit code is 0. That way we know that fetching the app information didnt fail and that there is information associated to the app. Not sure if it is a bit redundant tho 🤔